Document teleport-operator (#14965)
gravitational/teleport: docs/pages/setup/helm-reference/teleport-cluster.mdx

  </TabItem>
</Tabs>

## `installCRDs`

| Type | Default value | Can be used in `custom` mode? |
| - | - | - |
| `bool` | `false`| ✅ |

> CRDs are required by the teleport-operator and are installed by default when `operator.enabled` is true.

Reviewer: Is "teleport-operator" the way we're standardizing this product name?

> `installCRDs` overrides this behaviour and allows users to explicit whether to deploy Teleport CRDs.

Reviewer: ```suggestion
`installCRDs` overrides this behavior and allows users to indicate whether to deploy Teleport CRDs.
```
I think we're converging on US spelling

Unless you are deploying multiple `teleport-cluster` Helm releases in the same Kubernetes cluster or installing

> the CRDs on your own you should not have to set this value.

Reviewer: When would you want to set this value?

Author: When deploying multiple `teleport-cluster` releases in the same Kubernetes cluster

> <Admonition type="note">

Reviewer: We should break the text out of this Admonition box and move it higher up in the explanation of this field. Otherwise, we need to navigate to the end of this reference entry before we find out why this field exists.

Author: done

>   CRDs are not namespace-scoped resources, they can be installed only once in a cluster.

Reviewer: ```suggestion
  CRDs are not namespace-scoped resources—they can be installed only once in a cluster.
```

  If several releases of the `teleport-cluster` chart are deployed in the same Kubernetes cluster, only one
  release should have `installCRDs` enabled.
</Admonition>

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  installCRDs: true
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set installCRDs=true
  ```
  </TabItem>
</Tabs>

## `operator`

### `operator.enabled`

| Type | Default value | Can be used in `custom` mode? |
| - | - | - |
| `bool` | `false`| ✅ |

`operator.enabled` controls whether to deploy the Teleport Kubernetes Operator as a side-car.

> Enabling the operator will also deploy the Teleport CRDs in the Kubernetes clusters.

Reviewer: Which Kubernetes clusters?

Author: typo, it should not be plural

If you are deploying multiple releases of the Helm chart in the same cluster you can override this behavior with
[`installCRDs`](#installCRDs).

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  operator:
   enabled: true
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set operator.enabled=true
  ```
  </TabItem>
</Tabs>

### `operator.image`

| Type | Default value | Can be used in `custom` mode? |
| - | - | - |
| `string` | `quay.io/gravitational/teleport-operator`| ✅ |

`operator.image` sets the Teleport Operator container image used for Teleport pods in the cluster.

> You can override this to use your own Teleport Operator image rather than a Teleport-published image.

Reviewer: We should be consistent with how we capitalize "Operator".

Author: Done, every instance is aligned with `Teleport Kubernetes Operator`

This setting requires [`operator.enabled`](#operatorenabled).

<Tabs>
  <TabItem label="values.yaml">
  ```yaml
  operator:
    image: my.docker.registry/teleport-operator-image-name
  ```
  </TabItem>
  <TabItem label="--set">
  ```code
  $ --set operator.image=my.docker.registry/teleport-operator-image-name
  ```
  </TabItem>
</Tabs>

### `operator.resources`

| Type | Default value | Can be used in `custom` mode? |
| - | - | - |
| `object` | `{}` | ✅  |

> [Kubernetes reference](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/)

Reviewer: We should add some context for this link—right now it seems out of place until I mouse over it.

Author: Done

> Resource requests/limits which should be configured for the Teleport Operator container.

Reviewer: What does "should be" mean here? That this entry recommends that the reader configure these options?

Author: Rephrased

  </TabItem>
</Tabs>

## `installCRDs`

| Type | Default value | Can be used in `custom` mode? |
| - | - | - |
| `bool` | `false`| ✅ |

> CRDs are not namespace-scoped resources—they can be installed only once in a cluster.

Reviewer 2: ```suggestion
CRDs are not namespace-scoped resources - they can be installed only once in a cluster.
```

| - | - | - |
| `object` | `{}` | ✅  |

See the [Kubernetes resource](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/)
documentation.

> It is recommended to set Resource requests/limits for each container based on their observed usage.

Reviewer: ```suggestion
It is recommended to set resource requests/limits for each container based on their observed usage.
```
